api/analyzedphenotypes.systemvars.api.inc

```diff
@@ -0,0 +1,206 @@
+      // eg. LENS: cv - analyzedphenotypes_systemvar_lens_cv.
+      //           db - analyzedphenotypes_systemvar_lens_db.
+      //           on - analyzedphenotypes_systemvar_lens_ontology.
+      $result = chado_query(
+        "SELECT genus FROM {organism} GROUP BY genus ORDER BY genus ASC"
+      )
+      ->fetchCol();
+
+      foreach($result as $genus) {
+        $genus = strtolower(preg_replace('/\s+/', '_', $genus));
+
+        foreach($suffix as $term) {
+          // cv, db and on.
+          // Calling 'ap_ + genus' = ap_lens returns Genus lens cv, db and ontology variables.
+          $systemvars[$set]['ap_' . $genus][$term] = $var_prefix . $genus . '_' . $term;
+        }
+      }
+    }
+    else {
+      // Other terms.
+      // eg. For term METHOD: analyzedphenotypes_systemvar_method.
+      foreach($suffix as $term) {
+        $systemvars[$set]['ap_' . $term] = $var_prefix . $term;
+      }
+    }
+  }
+
+  return $systemvars;
+}
+
+
+///////
+
+
+/**
+ * CONTROL/SWITCHBOARD.
+ * Pass a request to a corresponding function.
+ * @TODO: leave out when api reorganize/update is complete and use functions below.
+ */
+function analyzedphenotypes_systemvariables($property = null, $parameter = null) {
+  $systemvars = null;
+
+  if (is_null($property)) {
+    // Default.
+    $property = 'default';
+  }
+
+  // Process request.
+  // Default to all system variables.
+  $command = array(
+    'default' => array(
+      'func'  => 'ap_define_variablenames',
+      'param' => null,
+      'opts'  => null,
+    ),
+    'options' => array(
+      'func'  => 'ap_get_systemvariable',
+      'param' => array('variablename' => 'allvars'),
+      'opts'  => array('set' => 'options'),
+    ),
+    'cvdbon' => array(
+      'func'  => 'ap_get_systemvariable',
+      'param' => array('variablename' => 'allvars'),
+      'opts'  => array('set' => 'cvdbon'),
+    ),
+    'terms' => array(
+      'func'  => 'ap_get_systemvariable',
+      'param' => array('variablename' => 'allvars'),
+      'opts'  => array('set' => 'terms'),
+    ),
+    'genus_vars' => array(
+      'func'  => 'ap_get_systemvariable',
+      'param' => array('variablename' => $parameter['genus']),
+      'opts'  => array('set' => 'cvdbon', 'suffix' => 'allsuffix'),
+    ),
+  );
+
+  // Function name, parameter (project name key or name), additional options.
+  return call_user_func($command[$property]['func'], $command[$property]['param'], $command[$property]['opts']);
+}
+
+
+///////
+
+
+/**
+ * FUNCTION:
+ * Fetch system variable name.
+ * Dependencies: ap_define_variablenames().
+ *
+ * @param $identifiers
+ *   An array with the following keys:
+ *     - variablename : term, option or genus keyword unique to a system variable.
+ * @param $options
+ *   An array to be used as additional options to be performed to returned result.
+ *     - set : the set a term belongs to. Default to CV, DB and ON (ontology) set.
+ *       allvars - all variables in a set.
```

Comment:
Based on the code it looks like `$identifiers = array('variablename' => 'allvars')` is how you retrieve all variables in a set. If so, this documentation is confusing since "allvars" is not a "set".